nullaway/src/test/java/com/uber/nullaway/Java8Tests.java

```diff
@@ -47,4 +47,24 @@ public void methodReferenceOnNullableVariable() {
             "}")
         .doTest();
   }
+
+  @Test
+  public void testNullableLambdaParamTypeUse() {
```

Comment:
Do we have the true positive case test somewhere? E.g. something showing that:

```
NonNullParamFunctionTypeUse n3 = (@Nullable Object x) -> (x == null) ? \"null\" : x.toString();",
```

Fails if `NonNullParamFunctionTypeUse` is equivalent to `NonNullParamFunctionTypeUse` but without `@Nullable` on `takeVal`.

Also, how about assigning a lambda that doesn't check for null?

Feel free to ignore if these tests already exist somewhere (they well might), just checking since this test below implies those cases too.

Reply:
I think those tests exist here:

https://github.com/uber/NullAway/blob/master/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayJava8NegativeCases.java
https://github.com/uber/NullAway/blob/master/nullaway/src/test/resources/com/uber/nullaway/testdata/NullAwayJava8PositiveCases.java

The key difference in the new unit test is that `@Nullable` is a type-use annotation rather than a declaration annotation, and with our new logic we initially weren't reading it out correctly.  I added a comment in a59a7a8402e489583870e1c9a5cb4e6200ab86b0